File: sketch/src/main/java/com/github/panpf/sketch/util/pool/LruPoolStrategy.java
package com.github.panpf.sketch.util.pool;

import android.graphics.Bitmap;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public interface LruPoolStrategy {

    void put(Bitmap bitmap);

    @Nullable
    Bitmap get(int width, int height, Bitmap.Config config);

    boolean exist(int width, int height, Bitmap.Config config);

    boolean exist(Bitmap bitmap);

    @Nullable
    Bitmap removeLast();

    @NonNull
    String logBitmap(@NonNull Bitmap bitmap);

    @NonNull
    String logBitmap(int width, int height, @Nullable Bitmap.Config config);

    int getSize(@NonNull Bitmap bitmap);
}

File: sketch-core/src/main/kotlin/com/github/panpf/sketch/util/pool/LruPoolStrategy.kt
package com.github.panpf.sketch.util.pool

import android.graphics.Bitmap
import android.graphics.Bitmap.Config

interface LruPoolStrategy {

    fun put(bitmap: Bitmap)

    fun get(width: Int, height: Int, config: Config): Bitmap?

    fun exist(width: Int, height: Int, config: Config): Boolean

    fun removeLast(): Bitmap?

    fun logBitmap(bitmap: Bitmap): String

    fun logBitmap(width: Int, height: Int, config: Config): String

    fun getSize(bitmap: Bitmap): Int
}